gitbook-docs/market/user-guides/proof-producer.md

```diff
@@ -1,82 +1,88 @@
 
-```
-python3 scripts/ask_tools.py get --key <key of the ask> 
+```bash
+python3 scripts/proposal_tools.py get --key <key of the proposal> 
 ```
 
-Ask's status 'processing' means that the ask was matched with a bid. Now it is time to generate a proof for the proof producer.
+If the proposal's status is `processing`, it means that the proposal was matched with the request.
+Now it's time for the proof producer to generate the proof.
 
-First of all, the proof producer needs circuit definition:
-
-```
+First of all, the proof producer needs statement definition:
 
+```bash
 python3 scripts/statement_tools.py get --key <key of the statement> -o <output file> 
-
 ```
 
-Next, public input of the bid:
+Next, public input of the request:
 
-```
-python3 scripts/public_input_get.py --key <bid key> -o <output file path> 
+```bash
+python3 scripts/public_input_get.py --key <request key> -o <output file path> 
 ```
 
-## Proof Generation
+## Generate a proof 
 
-Execute the below to generate proof:
-
-```
+To generate proof execute the following:
 
+```bash
 cd build
-./bin/proof-generator/proof-generator --proof_out=<output file> --circuit_input=<statement from Proof Market> --public_input=<public input from Proof Market>
+   ./bin/proof-generator/proof-generator --proof_out=<output file> \
+   --circuit_input=<statement from Proof Market> --public_input=<public input from Proof Market>
 # Or using the multithreaded version
-./bin/proof-generator/proof-generator-mt --proof_out=<output file> --circuit_input=<statement from Proof Market> --public_input=<public input from Proof Market> --smp=<number of threads> ----shard0-mem-scale=<scale>
-
+   ./bin/proof-generator/proof-generator-mt --proof_out=<output file> \
+   --circuit_input=<statement from Proof Market> --public_input=<public input from Proof Market> \
+   --smp=<number of threads> ----shard0-mem-scale=<scale>
 ```
 
-For multithreaded versions, the following flags warrant discussion as the computation of proof requires temporary space. This is always allocated to `core0`/`shard0.`
+Proof computation requires temporary space, always allocated to `core0`/`shard0`.
+The following flags require attention for multithreaded use:
 
-* `smp` : Number of threads. This is ideally the number of CPU cores on the system. ex: If you have a 16 core CPU &  16 GB RAM & set `smp` to 16. Each core gets access to 1 GB of RAM. Thus `core0`/`shard0` will have access to 1 GB of RAM.
-* `shard0-mem-scale`: Weighted parameter (`weight`) to reserve memory for `shard0.`This is a natural number which allows for `shard0` to have more access to RAM in comparison to others. ex: If you have a 16 core CPU &  16 GB RAM & set `smp` to 8 and set `shard0-mem-scale` to 9.
+* `smp` - number of threads.
+  This is ideally the number of CPU cores on the system.
+  If you have a 16-core CPU & 16 GB RAM, you can set `smp` to 16.
+  Each core gets access to 1 GB of RAM, so `core0`/`shard0` will have access to 1 GB of RAM.
```

Comment:
There's a neat trick that we could recommend:

```
# Linux
--smp $(nproc)

# macOS
--smp $(sysctl -n hw.logicalcpu)
```
See for example:
* https://github.com/NilFoundation/zkllvm/blob/master/.github/workflows/build_linux.yml#L105
* https://github.com/NilFoundation/zkllvm/blob/master/.github/workflows/build_macos.yml#L77

Reply:
Ok, but let's wait for Ilya to answer, if the multithreaded version isn't ready yet, maybe we'll have to remove this section

Reply:
Yep, since multithread processing is not supported yet, I'm removing this part